rllib/core/rl_module/tf/tf_rl_module.py

```diff
@@ -0,0 +1,62 @@
+from typing import Any, Mapping
+
+from ray.rllib.core.rl_module import RLModule
+from ray.rllib.core.rl_module.marl_module import MultiAgentRLModule
+from ray.rllib.utils.annotations import override
+from ray.rllib.utils.framework import try_import_tf
+
+
+_, tf, _ = try_import_tf()
+
+
+class TFRLModule(RLModule, tf.keras.Model):
+    def __init__(self, config: Mapping[str, Any]) -> None:
+        tf.keras.Model.__init__(self)
+        RLModule.__init__(self, config)
+
+    @override(tf.keras.Model)
+    def call(self, batch: Mapping[str, Any], **kwargs) -> Mapping[str, Any]:
+        """forward pass of the module.
+
+        This is aliased to forward_train because Torch DDP requires a forward method to
+        be implemented for backpropagation to work.
+        """
+        return self.forward_train(batch, **kwargs)
+
+    @override(RLModule)
+    def get_state(self) -> Mapping[str, Any]:
+        return self.get_weights()
```

Comment:
Can we add some comment here on how this works b/c of the underlying keras Model?

Reply:
To be frank im not sure how it works. I assume that it takes the trace of the tf model and collects those parameters.